apps/sms/views/shared/test/unit/activity_handler_test.js

```diff
@@ -113,211 +122,196 @@ suite('ActivityHandler', function() {
-          id: 'attached-files-too-large',
-          args: { n: 5, mmsSize: '0' }
-        });
+      sinon.assert.calledWith(Utils.alert, {
+        id: 'attached-files-too-large',
+        args: { n: 5, mmsSize: '0' }
+      });
+
+      Utils.alert.lastCall.returnValue.then(() => {
+        sinon.assert.calledWithExactly(ActivityClient.postResult);
+        sinon.assert.notCalled(Navigation.toPanel);
       }).then(done, done);
     });
 
-    test('Should append images even when they are big', function(done) {
-      shareActivity.source.data.blobs = [
-        new Blob(['test'], { type: 'image/jpeg' }),
-      ];
+    test('Should append images even when they are big', function() {
+      activityData.blobs = [new Blob(['test'], { type: 'image/jpeg' })];
 
       Settings.mmsSizeLimitation = 1;
-      this.sinon.spy(Compose, 'append');
 
-      MockNavigatormozSetMessageHandler.mTrigger('activity', shareActivity);
+      ActivityClient.on.withArgs('share-activity-request').yield(
+        activityData
+      );
 
-      panelPromise.then(function() {
-        sinon.assert.called(Compose.append);
-        sinon.assert.notCalled(Utils.alert);
-      }).then(done, done);
+      sinon.assert.notCalled(Utils.alert);
+
+      sinon.assert.calledWith(Navigation.toPanel, 'composer', {
+        activity: {
+          body: [sinon.match.instanceOf(Attachment)],
+          number: null
+        }
+      });
     });
 
-    test('Should append vcard attachment', function(done) {
-      shareActivity.source.data.blobs = [
-        new Blob(['test'], { type: 'text/x-vcard' }),
-      ];
+    test('Should append vcard attachment', function() {
+      activityData.blobs = [new Blob(['test'], { type: 'text/x-vcard' })];
 
       this.sinon.spy(Compose, 'append');
 
-      MockNavigatormozSetMessageHandler.mTrigger('activity', shareActivity);
+      ActivityClient.on.withArgs('share-activity-request').yield(
+        activityData
+      );
 
-      panelPromise.then(function() {
-        sinon.assert.called(Compose.append);
-      }).then(done, done);
-    });
+      sinon.assert.notCalled(Utils.alert);
 
-    test('share message should set the current activity', function(done) {
-      MockNavigatormozSetMessageHandler.mTrigger('activity', shareActivity);
-      panelPromise.then(function() {
-        assert.isTrue(ActivityHandler.isInActivity());
-      }).then(done, done);
+      sinon.assert.calledWith(Navigation.toPanel, 'composer', {
+        activity: {
+          body: [sinon.match.instanceOf(Attachment)],
+          number: null
+        }
+      });
     });
 
     test('Appends URL to the Compose field for activity with URL data type',
-      function(done) {
-      this.sinon.stub(Compose, 'append');
-      var shareURLActivity = {
-        source: {
-          name: 'share',
-          data: {
-            type: 'url',
-            url: 'test_url'
-          }
-        },
-        postResult: sinon.stub()
+      function() {
```

Comment:
nit: I now think the "function() {" part should be 2 pixels to the left so that we clearly see where the function code starts.